docs/src/modules/ROOT/pages/optimization-algorithms/move-selector-reference.adoc

```diff
@@ -0,0 +1,789 @@
+
+Simplest configuration:
+
+[source,xml]
+----
+    <subListChangeMoveSelector/>
+----
+
+Advanced configuration:
+
+[source,xml]
+----
+    <subListChangeMoveSelector>
+      ... <!-- Normal selector properties -->
+      <selectReversingMoveToo>true</selectReversingMoveToo>
+      <subListSelector id="subListSelector1">
+        <valueSelector>
+          ...
+        </valueSelector>
+        <minimumSubListSize>2</minimumSubListSize>
+        <maximumSubListSize>6</maximumSubListSize>
+      </subListSelector>
+    </subListChangeMoveSelector>
+----
+
+[#subListSwapMoveSelector]
+=== `SubListSwapMoveSelector`
+
+A _subList_ is a sequence of elements in a specific entity's list variable between `fromIndex` and `toIndex`.
+The `SubListSwapMoveSelector` selects a left subList by selecting a left entity and the left subList's `fromIndex` and `toIndex`.
+Then it selects a right subList by selecting a right entity and the right subList's `fromIndex` and `toIndex`.
+Selecting these parameters results in a `SubListSwapMove` that swaps the right and left subLists between right and left entities.
+
+Simplest configuration:
+
+[source,xml]
+----
+    <subListSwapMoveSelector/>
+----
+
+Advanced configuration:
+
+[source,xml]
+----
+    <subListSwapMoveSelector>
+      ... <!-- Normal selector properties -->
+      <selectReversingMoveToo>true</selectReversingMoveToo>
+      <subListSelector id="subListSelector1">
+        <valueSelector>
+          ...
+        </valueSelector>
+        <minimumSubListSize>2</minimumSubListSize>
+        <maximumSubListSize>6</maximumSubListSize>
+      </subListSelector>
+    </subListSwapMoveSelector>
+----
+
+[#kOptListMoveSelector]
+=== `KOptListMoveSelector`
+
+The `KOptListMoveSelector` considers the list variable to be
+a graph whose edges are the consecutive elements of the list
+(with the last element being consecutive to the first element).
+A `KOptListMove` selects an entity, remove `k` edges from its list variable, and add `k` new edges from the removed edges' endpoints.
+This move may reverse segments of the graph.
+
+image::optimization-algorithms/move-selector-reference/koptMove.png[align="center"]
+
+Simplest configuration:
+
+[source,xml]
+----
+    <kOptListMoveSelector/>
+----
+
+Advanced configuration:
+
+[source,xml]
+----
+    <kOptListMoveSelector>
+      ... <!-- Normal selector properties -->
+      <minimumK>2</minimumK>
+      <maximumK>4</maximumK>
+    </kOptListMoveSelector>
+----
+
+[#listRuinRecreateMoveSelector]
+=== `ListRuinRecreateMoveSelector`
+
+The `ListRuinRecreateMove` selects a subset of values, and removes them from their list variables.
+Then it runs a construction heuristic to assign them again.
+If xref:using-timefold-solver/modeling-planning-problems.adoc#planningListVariableAllowingUnassigned[unassigned values] are allowed,
+it may leave them unassigned.
+
+This coarse-grained move is useful to help get the solver out of a local optimum.
```

Comment:
´´´suggestion
This coarse-grained move is useful to help the solver to escape from a local optimum.
´´´